src/mainwindow.cpp

```diff
@@ -181,6 +181,21 @@ void MainWindow::InitData()
     emit requestNodesTree();
 }
 
+/*!
+ * \brief Toggles visibility of the main window upon system tray activation
+ * \param reason The reason the system tray was activated
+ */
+void MainWindow::onSystemTrayIconActivated(const QSystemTrayIcon::ActivationReason reason)
+{
+    // We disable this functionality altogether on macOS because a single left/right
+    // mouse click on the tray icon displays its context menu, which is a bit awkward.
```

Comment:
I noticed there is also `QSystemTrayIcon::MiddleClick`, maybe that could be an alternative for macOS?

Alternatively, since the menu only has "show" and "quit" actions anyway, we can remove the menu entirely on macOS and just keep the toggle behavior. Quitting on macOS anyway has its own action in the application menu and by Command-Q.

Reply:
Yes, I guess we don't really need the context menu on macOS when keeping the toggle behavior.

Reply:
Let me know when I can test this.